scipy/ndimage/tests/test_morphology.py

```diff
@@ -283,34 +317,41 @@ def test_distance_transform_bf06(self, dtype):
                      [0, 1, 2, 3, 4, 5, 6, 7, 8]]]
         assert_array_almost_equal(ft, expected)
 
-    def test_distance_transform_bf07(self):
+    def test_distance_transform_bf07(self, xp):
+        if is_cupy(xp):
+            pytest.xfail("CuPy does not have distance_transform_bf.")
+
         # test input validation per discussion on PR #13302
-        data = np.array([[0, 0, 0, 0, 0, 0, 0, 0, 0],
-                         [0, 0, 0, 0, 0, 0, 0, 0, 0],
-                         [0, 0, 0, 1, 1, 1, 0, 0, 0],
-                         [0, 0, 1, 1, 1, 1, 1, 0, 0],
-                         [0, 0, 1, 1, 1, 1, 1, 0, 0],
-                         [0, 0, 1, 1, 1, 1, 1, 0, 0],
-                         [0, 0, 0, 1, 1, 1, 0, 0, 0],
-                         [0, 0, 0, 0, 0, 0, 0, 0, 0],
-                         [0, 0, 0, 0, 0, 0, 0, 0, 0]])
+        data = xp.asarray([[0, 0, 0, 0, 0, 0, 0, 0, 0],
+                           [0, 0, 0, 0, 0, 0, 0, 0, 0],
+                           [0, 0, 0, 1, 1, 1, 0, 0, 0],
+                           [0, 0, 1, 1, 1, 1, 1, 0, 0],
+                           [0, 0, 1, 1, 1, 1, 1, 0, 0],
+                           [0, 0, 1, 1, 1, 1, 1, 0, 0],
+                           [0, 0, 0, 1, 1, 1, 0, 0, 0],
+                           [0, 0, 0, 0, 0, 0, 0, 0, 0],
+                           [0, 0, 0, 0, 0, 0, 0, 0, 0]])
         with assert_raises(RuntimeError):
             ndimage.distance_transform_bf(
                 data, return_distances=False, return_indices=False
             )
 
     @pytest.mark.parametrize('dtype', types)
-    def test_distance_transform_cdt01(self, dtype):
+    def test_distance_transform_cdt01(self, dtype, xp):
+        dtype = getattr(xp, dtype)
+        if is_cupy(xp):
+            pytest.xfail("CuPy does not have distance_transform_cdt.")
```

Comment:
we really need an `@xfail_xp_backends` - I suppose this can merge without it, though

Reply:
Let's keep it to a follow-up clean-up PR. This @-decorator would play nice with `@parametrize`, so it's best separated out.